editor/src/clj/editor/defold_project.clj

```diff
@@ -858,43 +1261,57 @@
           load-progress-span 3
           total-progress-span (+ read-progress-span load-progress-span)
           total-progress (progress/make "" total-progress-span 0)
-          deleted-node-ids (set (:delete plan))
-
-          old-resource-node-ids-by-proj-path
-          (if (coll/empty? deleted-node-ids)
-            old-nodes-by-path
-            (persistent!
-              (reduce
-                (fn [old-resource-node-ids-by-proj-path [proj-path old-node-id]]
-                  (if (contains? deleted-node-ids old-node-id)
-                    (dissoc! old-resource-node-ids-by-proj-path proj-path)
-                    old-resource-node-ids-by-proj-path))
-                (transient old-nodes-by-path)
-                old-nodes-by-path)))
+          deleted-node-id? (set (:delete plan))
+
+          old-node-id->old-node-state
+          (fn old-node-id->old-node-state [old-node-id]
+            ;; Returns the state of the supplied old-node-id as it existed in
+            ;; the graph before the resource change plan was executed.
+            (if (resource-node/loaded? old-basis old-node-id)
+              :loaded
+              :not-loaded))
```

Comment:
At one point, we also had `:not-found`, but I didn't change it back to a boolean. I kind of like the clarity of keywords.